internal/internal_workflow_testsuite.go

```diff
@@ -2970,3 +3150,161 @@ func (h *testNexusOperationHandle) cancel() {
 		}, false)
 	}()
 }
+
+type testNexusHandler struct {
+	nexus.UnimplementedHandler
+
+	env      *testWorkflowEnvironmentImpl
+	opHandle *testNexusOperationHandle
+	handler  nexus.Handler
+}
+
+func newTestNexusHandler(
+	env *testWorkflowEnvironmentImpl,
+	opHandle *testNexusOperationHandle,
+) (nexus.Handler, error) {
+	reg := nexus.NewServiceRegistry()
+	for _, service := range env.registry.nexusServices {
+		if err := reg.Register(service); err != nil {
+			return nil, fmt.Errorf("failed to register nexus service '%v': %w", service, err)
+		}
+	}
+	handler, err := reg.NewHandler()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create nexus handler: %w", err)
+	}
+	return &testNexusHandler{
+		env:      env,
+		opHandle: opHandle,
+		handler:  handler,
+	}, nil
+}
+
+func (r *testNexusHandler) StartOperation(
+	ctx context.Context,
+	service string,
+	operation string,
+	input *nexus.LazyValue,
+	options nexus.StartOperationOptions,
+) (nexus.HandlerStartOperationResult[any], error) {
+	s, ok := r.env.registry.nexusServices[service]
+	if !ok {
+		panic(fmt.Sprintf(
+			"nexus service %q is not registered with the TestWorkflowEnvironment",
+			service,
+		))
+	}
+	op, ok := s.GetOperation(operation)
+	if !ok {
+		panic(fmt.Sprintf(
+			"nexus operation %q is not registered in service %q with the TestWorkflowEnvironment",
+			operation,
+			service,
+		))
+	}
+	fn, _ := reflect.TypeOf(op).MethodByName("Start")
+	inputType := fn.Type.In(2)
+	ptr := reflect.New(inputType)
+	if err := input.Consume(ptr.Interface()); err != nil {
+		panic(fmt.Sprintf("mock of ExecuteNexusOperation failed to deserialize input"))
+	}
+
+	// rebuild the input as *nexus.LazyValue
```

Comment:
Why?

Reply:
Because, after I call `input.Consume` on L3208, it closed the reader, so it can't call again, which is needed if the user didn't mock the operation.

Reply:
Well, actually, I can see that the `reader` isn't really being used at this moment, so maybe I could just skip this...

Reply:
Yeah, you can avoid this for now since we control both implementations. Might want to just put a comment here but either way it's not critical.

Reply:
I'd rather leave as it to avoid forgetting about this later if things change.